### DISCUSSION TITLE
:star: v2로 넘어가며 개선되어야 할 부분 논의

### DISCUSSION THREAD

**Discussion by Gyaak** [Kernel360/F2-BAGUNI#201] [개발 토의]
# v2로 넘어가면서 v1에서 시간 및 설계이슈로 추가/수정 하지 못했던 것들을 추가 및 수정하려고 합니다.

조금이라도 불편한 점이 있었다면 자유롭게 의견 주세요
@dmdgpdi @kimminkyeu @enigsuss @sangwonsheep 

---

## 소셜 로그인 방식 수정
1. 프론트에서 oauth서버로 인증을 요청
2. 인증 완료 후 인증서버로 부터 authorization code 받음
3. 프론트에서 서버로 code 전송
4. 서버에서 받은 code로 oauth서버에 유저 정보 요청
5. 받은 유저정보를 기반으로 access_token과 refresh_token 생성하여 프론트로 전달

spring ouath client을 사용하지 않고 Filter를 이용해 구현하도록 수정..

---

## 폴더 및 픽의 중복 이름 허용
V1의 폴더 및 픽 중복이름 불가 정책은 사용자 경험에 좋지 않다고 판단되어
V2에서는 폴더/픽 이름 중복 허용으로 변경

---

## 익스텐션에서 최근 추가한 픽 목록 조회
가장 최근 추가한 픽 10개 조회
해당 목록해서 픽 수정 및 삭제 가능

---

## 픽 존재여부 확인을 위한 url로 픽 검색 필요
url을 파라미터로 받아 해당 url을 가진 픽이 존재하는지 여부 반환하는 api 추가 필요

---

**Comment by kimminkyeu** [comment]
### 회원 탈퇴 기능
- 추가 예정

**Comment by kimminkyeu** [comment]
### 쿼리 별도 로깅
AOP 써서 쿼리 추적이 필요할 것 같은 메소드에 @LogQuery 같은 커스텀 어노테이션 남겨서 별도 로그 남기기 
1. 호출 메소드 이름.
2. 발생한 쿼리.
3. 발생 빈도?

**Comment by kimminkyeu** [comment]
### 링크 관련 기능
> Save First, Watch Later
:eyes: 먼저 담고 확인은 나중에

#### 미확인 북마크 구분
1. 사용자는 내가 이 링크를 북마크만 했는지, 정말 이걸 봤는지 확인할 수 있어야 한다.
2. 만약 내가 북마크를 하고 확인하지 않았다면, 그 내용을 "안본 게시물"로 따로 볼 수 있으면 좋겠다.

#### 링크 추적 
Pick:`언제 담았는가?`  +  Link:`언제 봤는가?` 
위 두가지를 나눠서 추적한다.
##### Pick
- 생성일자 활용 --> DB에 있음
##### Link Proxy
- 해당 링크를 우리 사이트의 링크로 치환
- 비유효화된 링크를 스케쥴러로 체크한다.
- 이미지, 제목 등의 원문 정보를 최신으로 업데이트 한다. (수정이 발생할 수 있기 때문)
- 링크를 좀 이쁘게 만들어준다. (`너무 길면 짧게, 제목도 적당히....`)

**Reply by kimminkyeu** [reply]
### 로그로 이벤트를 쌓는 거가 있기 때문에, 그거랑 구분해서 해야 함
그래서 링크 테이블에 이걸 넣으면 안된다.

**Reply by kimminkyeu** [reply]
봤냐 안봤냐는 링크에서  빼야 함

**Reply by kimminkyeu** [reply]
### 추가적으로, 링크뷰이벤트가 픽뷰이벤트로 변경 될 것.
왜냐면 유저마다의 로그 추적이기 때문이다.

**Reply by kimminkyeu** [reply]
### URL 치환 부분만 여기 들어가면 됨.
